ariadne/cardano/src/Ariadne/MainTemplate.hs

```diff
@@ -162,23 +163,23 @@ initializeEverything MainSettings {..}
     initAction :: IO ()
     initAction = walletInitAction
 
+    postStartupAction :: IO ()
+    postStartupAction = postInitAction
+
     serviceAction :: IO ()
     serviceAction =
-      raceWithUpdateCheckAction $
-      cardanoAction
+      raceWithUpdateCheckAction cardanoAction
 
     mainAction :: IO ()
     mainAction = do
       initAction
-
       -- Spawn backend actions in async thread, then run ui action in the main thread
       -- This is needed because some UI libraries (Qt) insist on livng in the main thread
-      withAsync serviceAction $ \serviceThread -> do
-        -- Make custom link to service thread for UI apps.
-        -- It is going to call msPutBackendErrorToUI and rethrow exception, if something goes wrong
-        linkUI serviceThread $ msPutBackendErrorToUI uiFace
-        uiAction
-
+      withAsync (concurrently postStartupAction serviceAction) $ \serviceThread -> do
+          -- Make custom link to service thread for UI apps.
```

Comment:
Well, I still don't like that you changed the indentation here (and made it inconsistent with the rest of this function).